src/pages/certificates/certificates-page.js

```diff
@@ -0,0 +1,138 @@
+import Status from './status/status';
+import LoadingBar from '../../components/loading-bar/loading-bar';
+import Certificate from './certificate/certificate';
+import { useCommonStyles } from '../common.styles';
+import { getAllCertificates } from './operations/certificate.queries';
+import { getUsers } from '../../redux/users/users.actions';
+import { config } from '../../configs';
+
+const { routes } = config;
+const pathToCreateCertificatesPage = routes.pathToCreateCertificates;
+const pageTitle = config.titles.certificatesPageTitles.mainPageTitle;
+const tableTitles = config.tableHeadRowTitles.certificates;
+const { CREATE_CERTIFICATE_TITLE } = config.buttonTitles;
+const { ACTIVE_STATUS, USED_STATUS, EXPIRED_STATUS } = config.statuses;
+
+const transformDate = (date) => {
+  const exactDate = new Date(date);
+  return exactDate.toLocaleString('en-US', {
+    month: '2-digit',
+    day: '2-digit',
+    year: 'numeric'
+  });
+};
+
+const checkStatus = (active, used) => {
+  if (active && !used) {
+    return ACTIVE_STATUS;
+  }
+  if (!active && used) {
+    return USED_STATUS;
+  }
+  if (!active && !used) {
+    return EXPIRED_STATUS;
+  }
+};
+
+const CertificatesPage = () => {
+  const commonStyles = useCommonStyles();
+  const dispatch = useDispatch();
+  const { loading: certificatesLoading, data: certificates } =
+    useQuery(getAllCertificates);
+  const certificatesList = certificates?.getAllCertificates || {};
+
+  useEffect(() => {
+    dispatch(getUsers({}));
+  }, []);
+
+  const { list: usersList, loading: usersLoading } = useSelector(
+    ({ Users }) => ({
+      list: Users.list,
+      loading: Users.userLoading
+    })
+  );
+
+  const setUser = (id) => {
+    if (id && usersList.length) {
+      for (const user of usersList) {
+        if (user._id === id) {
+          return `${user.firstName} ${user.lastName}`;
+        }
+      }
+    } else return '';
+  };
+
+  const editCertificate = () => {
+    // TODO
+  };
+
+  const deleteCertificate = () => {
+    // TODO
+  };
+
+  const certificateItems = certificatesList.items
+    ? certificatesList.items.map((certificate) => (
+        <TableContainerRow
+          key={certificate._id}
+          number={certificate.name}
+          createdBy={<Certificate name={setUser(certificate.createdBy._id)} />}
+          price={`${certificate.value} грн`}
+          status={
+            <Status
+              status={checkStatus(certificate.isActivated, certificate.isUsed)}
+            />
+          }
+          date={
+            certificate.isActivated
+              ? `${transformDate(certificate.dateStart)} - ${transformDate(
+                  certificate.dateEnd
+                )}`
+              : '-'
+          }
+          deleteHandler={deleteCertificate}
+          editHandler={editCertificate}
+          showAvatar={false}
+        />
+      ))
+    : null;
+
+  if (certificatesLoading && usersLoading) {
+    return <LoadingBar />;
+  }
```

Comment:
Maybe make conditional rendering on TableContainerGenerator . Is there a reason to hide <Button data-cy='add-certificate' component ?

Reply:
There is no specific reason this is just how other pages render the loading bar.